src/main/frontend/handler/editor.cljs

```diff
@@ -3425,20 +3440,26 @@
                  (collapse-block! uuid))))))))))
 
 (defn- collapse-all!
-  []
-  (let [blocks-to-collapse
-        (->> (all-blocks-with-level {:collapse? true})
-             (filter (fn [b] (collapsable? (:block/uuid b)))))]
-    (when (seq blocks-to-collapse)
-      (doseq [{:block/keys [uuid]} blocks-to-collapse]
-        (collapse-block! uuid)))))
+  ([]
+   (collapse-all! nil))
+  ([block-id]
+   (collapse-all! block-id nil))
+  ([block-id max-level]
+   (let [blocks-to-collapse (all-blocks-with-level {:expanded? true :root-block block-id :max-level max-level})]
+     (when (seq blocks-to-collapse)
+       (doseq [{:block/keys [uuid]} blocks-to-collapse]
+         (collapse-block! uuid))))))
 
 (defn- expand-all!
```

Comment:
This function is private.

Reply:
@andelf Now collapse-all! and expand-all! are public functions